sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -48,13 +49,18 @@
 import org.apache.beam.sdk.transforms.DoFn;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.transforms.ParDo;
```

Comment:
We'll want to update https://github.com/apache/beam/blob/master/CHANGES.md to state that JmsIO is changing such that a value mapper is required. For users to get the existing behavior, they should invoke JmsIO.write().withValueMapper(new TextMessageMapper())

We should also highlight that writing to JmsIO is gaining the ability to map to multiple JMS message types and also that will support dynamic destinations.

Reply:
Was not it possible to avoid this breaking change? Like, add new functionality and keep a previous one  deprecated? 

Just tested the latest Beam 2.39.0 RC2 and my job with JmsIO.write() failed to compile without additional code changes.